.changeset/tidy-carrots-jump.md

```diff
@@ -0,0 +1,31 @@
+---
+'@astrojs/vercel': minor
+'@astrojs/node': minor
+'astro': patch
+---
+
+Add experimental support for i18n domain routing:
+
+```js
+// astro.config.mjs
+import { defineConfig } from "astro/config"
+export default defineConfig({
+    experimental: {
+        i18n: {
+            defaultLocale: "en",
+            locales: ["es", "en", "fr"],
+            routingStrategy: "domain",
+            domains: {
+                fr: "https://fr.example.com",
+                es: "https://example.es"
+            }
+        }
+    }
+})
+```
+
+When this routing strategy is enabled and you run `astro build` command, the URLs emitted by `getAbsoluteLocaleUrl` and `getAbsoluteLocaleUrlList` will be different:
+
+- `/fr/about` will become `https://fr.example.com/about`;
+- `/es/about` will become `https://example.es/about`;
+- `/en/about` will **stay** as is because we haven't configured any domain;
```

Comment:
```suggestion
With `routingStrategy: "domain"` configured, the URLs for your built site, including the URLs emitted by `getAbsoluteLocaleUrl()` and `getAbsoluteLocaleUrlList()`, will follow the pattern set for each locale individually.

For any `locales` not configured in `domains`, the fallback URLs created will follow `prefix-other-locales`, and create a `/[locale]/` path for all non-default languages.

For the above configuration:

- The file `/fr/about.astro` will create the URL `https://fr.example.com/about`;
- The file `/es/about.astro` will create the URL `https://example.es/about`;
- the file `/ja/about/astro` will create the URL `https://example.com/ja/about`.
- The file `/en/about.astro` will create the URL `https://example.com/about`.
```

So, I'm not sure whether this is true! But, I think adding `"ja"` to the list of locales and showing what happens would be very helpful. OR, specify that you are only allowed to not specify the defaultLocale, if that's the case. I think only the one example of "en" isn't enough to establish the pattern because it happens to be the default locale, so it could be a special case.

Reply:
Re 1.: good call out, I belive adding one more non-confiured locale in the example is perfect. 

Re 2: 
> I'm not sure whether you'd prefer to describe this as a "redirect" or "a file in this location produces this URL". Either is fine!

I prefer `"a file in this location produces this URL"` because it's closer to the actual behaviour. **We don't redirects**; we make assumptions, and developers must follow those assumptions if they want to have this feature.